Integrated bookdown documentation, documentation for project (#9)
PecanProject/web: README.md

### PEcAn People

The people page contains details of the PEcAn team. The image here is a representation of how the page should look like with placeholder values.

![PecanPople](screenshots/pecanPeople.png)

To add and modify information move to the `static/utils/contributors.js` file. It contains an array called `contributors` which contains JavaScript objects which are enclosed by `{}` and contain following fields-

![Contributors.js](screenshots/contributors.png)

In order to add a contributor, copy any object enclosed by `{}` and paste after the last object. Make sure two objects are separated by a `,`. Now one can easily edit information in the pasted object.
One could add information accordingly in the different fields. The information would be automatically updated after successful push.

> 

Author: The images are actually used by `src/components/CardList.js` component. It loops over the array we get from `static/utils/contributors.js` and renders a list of cards that we display at `people` page in the website.

        <li><a href="#adding-tutorials">Adding tutorials section</a></li>
        <li><a href="#blog">Blog</a></li>
      </ul>
    </li>
  </ol>
</details>

## Getting Started - Setting up the project for local development 

This section will help one to setup the project for local development.

### Requirements

- Node version >= 12.13.0
- yarn

This project has been built using [Docusaurus 2](https://docusaurus.io/), which requires Node version >=12.13.0, you can check this using `node -v`. [Install node](https://nodejs.org/en/download/).
Make sure yarn is installed. [Install yarn](https://classic.yarnpkg.com/en/docs/install/#debian-stable).

### Installation

```console
yarn install
```

### Local Development

```console
yarn start
```

This command starts a local development server and opens up a browser window. Most changes are reflected live without having to restart the server.

### Build

```console
yarn build
```

This command generates static content into the `build` directory and can be served using any static contents hosting service.

## Adding Content

This describes how to add or modify the Website's content.

> ### Feature Cards

Reviewer: ```suggestion
###  Add or Edit Feature Cards
```

Feature Cards on the website homepage highlight the key features of the PEcAn Project.

![Feature Card](screenshots/featureCard.png)

To modify the feature cards-

- Navigate to `src/components/HomepageFeatures.js`
- Navigate to `const FeatureList` inside `HomepageFeatures.js`

![FeatureList](screenshots/FeatureList.png)

Every item inside the `FeatureList` array contains a JavaScript object enclosed `{}` and separated by `,` . 
Each of these objects is a feature that is displayed on the homepage and contains a `title`, `Svg` and `description`. 
You can edit or remove feature cards and the changes will be reflected after the code passes build test. 
If you are adding or editing an image file, the image file must is present inside `static/img` directory, otherwise the compiler would return an error.
You should use SVG images when available.

> ### PEcAn People

Reviewer: ```suggestion
###  Add or Edit a Person (Contributors) 
```

The people page contains details of the PEcAn team. The image here is a representation of how the page should look like with placeholder values.

![PecanPople](screenshots/pecanPeople.png)

To add and modify information move to the `static/utils/contributors.js` file. 
It contains an array called `contributors` which contains JavaScript objects which are enclosed by `{}` and contain following fields-

![Contributors.js](screenshots/contributors.png)

In order to add a contributor, copy any object enclosed by `{}` and paste after the last object. Make sure two objects are separated by a `,`. Now one can easily edit information in the pasted object.
One could add information accordingly in the different fields. The information would be automatically updated after successful push.

> ### Blog

Reviewer: ```suggestion
### Add or Edit a Blog Post
```

2. In the top of your blog file add a yaml header.
3. Add at least `title`, `author` and `tags` yaml fields. 
  - These metadata fields help with search and navigation on the site. 
  - These and other yaml header options are described in the [blog header options](https://docusaurus.io/docs/blog#header-options) section of the Docusaurus documentation.   
4. writing your blog and save the file. 
5. any images can be put in `static/img/` and referenced using a relative path `static/img/myimage.png` 
6. build the website to make sure the blog shows up as expected 
7. After the new changes are merged into the main branch, the new blog post will appear on the website.

![YamlHeader](screenshots/yamlHeader.png)

<!-- instead of that screenshot, perhaps provide a code chunk here  -->

To learn more about markdown visit (https://guides.github.com/features/mastering-markdown/). 

## Screenshots

This section contains screenshots of website's homepage in light and dark theme.

![LightTheme](screenshots/light.png)

![DarkTheme](screenshots/dark.png)

## Contributing

This is an open source project and contributions are what make an open source community.

> Everyone is welcome to contribute to this project. The rules are simple:

Reviewer: ```suggestion
Everyone is welcome to contribute to this project. 
All involved are expected to follow the [PEcAn code of conduct](https://pecanproject.github.io/pecan-documentation/master/contributor-covenant-code-of-conduct.html) and contribute to a positive and inclusive environment.
```

> - Be polite, be patient.

Reviewer: ```suggestion
```

> - Be specific about your doubts, make sure you read this documentation carefully.

Reviewer: ```suggestion
```

just want to stick to the official code of conduct (also to avoid implying that contributors should RTFM before asking any questions)

First time contributors are welcome. Contributions can be very simple, make sure to check out beginner friendly issues, if present.

## Next Steps

This section contains future plans for the website. New contributors can use this section as a reference for planning their contributions.

### Optimising Documentation

At present the PEcAn's documentation is being served from (https://pecanproject.github.io/pecan-documentation/master/) using an iframe. While this saves a lot of overhead in converting Rmd files in (https://github.com/PecanProject/pecan/tree/develop/book_source) to md files so that docusaurus could render them, it also increases the loading time for documentation, it also looks less consistent.

> In future, it is suggested to find an optimised way to convert rmd files to md and integrate them via docusaurus's native documentation parser.

Reviewer: maybe create a new issue with details, and refer to it. But this should include next steps / outline of what needs to be done e.g. 

- automating build to md rather than html (not natively supported by bookdown)
- what are steps to integrate ... 
  - build md files to (what folder?)
  - change config files to ???
  - remove the current iframe approach (where?)

Author: Got it, adding a new issue is a good idea.

> ### Adding tutorials section

Reviewer: same as above
